File: 3279-alice-and-bob-playing-flower-game/alice-and-bob-playing-flower-game.java
class Solution {
    public long flowerGame(int n, int m) {
        long even_n = n / 2, odd_n = n - even_n;
        long even_m = m / 2, odd_m = m - even_m;
        return (even_n * odd_m)+ (odd_n * even_m);
    }
}

File: 3279-alice-and-bob-playing-flower-game/alice-and-bob-playing-flower-game.rs
impl Solution {
    pub fn flower_game(n: i32, m: i32) -> i64 {
        let even_n: i64 = (n as i64) / 2;
        let odd_n: i64 = (n as i64) - even_n;
        let even_m: i64 = (m as i64) / 2;
        let odd_m: i64 = (m as i64) - even_m;
        (even_n * odd_m) + (odd_n * even_m)
    }
}